sdk/test-utils/recorder-new/MIGRATION.md

```diff
@@ -0,0 +1,270 @@
+
+Here is an example:
+
+```ts
+const queueName = recorder.variable("queueName", "queue-${Math.floor(Math.random * 1000)}");
+// Assume that we have a client that has a createQueue method.
+client.createQueue(queueName);
+```
+
+In this example, the name of the queue used in the recording is randomized. However, in playback, instead of using the value passed into `recorder.variable`, the value will be retrieved from the recording file. This means that the name of the queue will be consistent between recording and playback.
+
+## Customizations on recordings
+
+A powerful feature of the legacy recorder was its `customizationsOnRecordings` option, which allowed for arbitrary replacements to be made to recordings. The new recorder's analog to this is the sanitizer functionality.
+
+### GeneralRegexSanitizer
+
+For a simple find/replace, a `GeneralRegexSanitizer` can be used. For example:
+
+```ts
+recorder.addSanitizers({
+  generalRegexSanitizers: [
+    {
+      regex: "find", // This should be a .NET regular expression as it is passed to the .NET proxy tool
+      value: "replace"
+    }
+    // add additional sanitizers here as required
+  ]
+});
+```
+
+This example would replace all instances of `find` in the recording with `replace`.
+
+### ConnectionStringSanitizer
+
+A `ConnectionStringSanitizer` can be used to strip all occurrences of a connection string from a recording. Its usage is very similar to the `GeneralRegexSanitizer`. For example:
+
+```ts
+recorder.addSanitizers({
+  connectionStringSanitizers: [
+    {
+      actualConnString: /* the actual connection string to be replaced, usually passed in as an environment variable */,
+      fakeConnString: /* a mock connection string to replace actualConnString with */,
+    },
+  ],
+});
+```
+
+### RemoveHeaderSanitizer
+
+`RemoveHeaderSanitizer` can be used to remove specific headers from the recordings as follows:
+
+```ts
+recorder.addSanitizers({
+  removeHeaderSanitizer: {
+    headersForRemoval: ["Header1", "Header2" /* ... */]
+  }
+});
+```
+
+Other sanitizers for more complex use cases are also available.
+
+## AAD and the new `NoOpCredential`
+
+The new recorder does not record AAD traffic at present. As such, tests with clients using AAD should make use of the new `@azure-tools/test-credential` package, installed as follows:
+
+```bash
+$ rush add --dev --caret -p @azure-tools/test-credential
+```
+
+This package provides a `NoOpCredential` implementation of `TokenCredential` which makes no network requests, and should be used in playback mode. The provided `createTestCredential` helper will handle switching between NoOpCredential in playback and ClientSecretCredential when recording for you:
+
+```ts
+const credential = createTestCredential();
+
+// You may now pass the created credential to your client.
```

Comment:
add one more line showing the credential being passed

Reply:
I left this out as it's fairly trivial and the exact code required to pass the credential will vary depending on the client.

Reply:
`new MyServiceClient(<endpoint>, credential)` should be good, just to remind them what a credential is.

Reply:
Alright